cli/internal/oauth/api/api.go

```diff
@@ -119,12 +121,26 @@ func (a API) WaitForDeviceToken(ctx context.Context, state State) (TokenResponse
 			}
 
 			return res, nil
-		case <-timeout:
+		case <-timeout.C:
 			return TokenResponse{}, ErrTimeout
 		}
 	}
 }
 
+// resetTimer is a helper function thatstops, drains and resets the timer.
+// This is necessary in go versions <1.23, since the timer isn't stopped +
+// the timer's channel isn't drained on timer.Reset.
+// See: https://go-review.googlesource.com/c/go/+/568341
```

Comment:
Should there be a `TODO` here to redo this when Go is upgraded?

Reply:
Oh, sure! The current code will be fine, but we'll be able to simplify this.

Reply:
Added